packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -139,287 +138,377 @@ function readFile(filename, encoding) {
-            if (pgEnableTags && object.description) {
-              const parsed = parseTags(object.description);
-              object.tags = parsed.tags;
-              object.description = parsed.text;
-            } else {
-              object.tags = {};
-            }
-          });
-        });
+      const extensionConfigurationClassIds = flatMap(
+        pgStructuresByKind.extension,
+        e => e.configurationClassIds
+      );
+      pgStructuresByKind.class.forEach(klass => {
+        klass.isExtensionConfigurationTable =
+          extensionConfigurationClassIds.indexOf(klass.id) >= 0;
+      });
 
-        const extensionConfigurationClassIds = flatMap(
-          pgStructuresByKind.extension,
-          e => e.configurationClassIds
-        );
-        pgStructuresByKind.class.forEach(klass => {
-          klass.isExtensionConfigurationTable =
-            extensionConfigurationClassIds.indexOf(klass.id) >= 0;
-        });
+      return pgStructuresByKind;
+    })
+  );
+}
 
-        return pgStructuresByKind;
-      })
-    );
+/**
+ * Verifies if the returned PostgresQL structures contain the schemas requested
+ *
+ * @param namespaces
+ * @param requestedSchemas
+ * @param throwIfMissing
+ */
+function verifySchemas(namespaces, requestedSchemas, throwIfMissing) {
+  //extract the schemas found in the database
+  const knownSchemas = namespaces.map(n => n.name);
+
+  //check to see if any of the requested schemas are not actually present in the database
+  const missingSchemas = requestedSchemas.filter(
+    s => knownSchemas.indexOf(s) < 0
+  );
+  if (missingSchemas.length) {
+    const errorMessage = `
+      You requested to use schema '${requestedSchemas.join("', '")}';
+      however we couldn't find some of those! Missing schemas are: '${missingSchemas.join(
+        "', '"
+      )}'
+    `;
+    if (throwIfMissing) {
+      throw new Error(errorMessage);
+    } else {
+      console.warn("⚠️ WARNING⚠️  " + errorMessage); // eslint-disable-line no-console
+    }
+  }
+}
 
-    //extract the schemas found in the database
-    const knownSchemas = introspectionResultsByKind.namespace.map(n => n.name);
-
-    //check to see if any of the requested schemas are not actually present in the database
-    const missingSchemas = schemas.filter(s => knownSchemas.indexOf(s) < 0);
-    if (missingSchemas.length) {
-      const errorMessage = `
-        You requested to use schema '${schemas.join("', '")}';
-        however we couldn't find some of those! Missing schemas are: '${missingSchemas.join("', '")}'
-      `;
-      if (pgThrowOnMissingSchema) {
-        throw new Error(errorMessage);
+/**
+ * Takes the result of getPGStructuresByKind and extracts the tags based on the
+ * smart commenting system exposed by the 'parseTags' function from utils
+ *
+ * @param pgStructuresByKind
+ * @param smartComments - True iff smart comments are enabled
+ */
+function parseTagsFromStructures(pgStructuresByKind, smartComments) {
```

Comment:
A more explicitly "mutating" name would be good (otherwise you'd expeect it to return); perhaps `addTagsToStructures` or `parseTagsIntoStructures` or `processTagsForStructures`?

Reply:
Good call.